src/controllers/EvidenceUploadController.ts

```diff
@@ -34,32 +39,94 @@ export class EvidenceUploadController extends BaseController<OtherReason> {
 
+    private async renderUploadError( appeal: Appeal, text: string ): Promise<void> {
+        const that = this;
+        const validationResult: ValidationResult = new ValidationResult( [
+            new ValidationError('file', text)
+        ]);
+
+        return await that.renderWithStatus(UNPROCESSABLE_ENTITY)(
+            that.template, {
+                ...this.prepareViewModelFromAppeal(appeal),
+                ...this.httpContext.request.body,
+                validationResult
+            }
+        );
+    }
+
     protected getExtraActionHandlers(): Record<string, FormActionHandler | FormActionHandlerConstructor> {
         const that = this;
         return {
             'upload-file': {
                 async handle(request: Request, response: Response): Promise<void> {
-                    await parseFormData(request, response);
-
-                    if (!request.file) {
-                        response.redirect(request.route.path);
-                        return;
-                    }
 
-                    const id = await that.fileTransferService.upload(request.file.buffer, request.file.originalname);
+                    const maxNumberOfFiles: number = Number(getEnvOrThrow('MAX_NUMBER_OF_FILES'));
 
                     const appeal: Appeal = request.session
                         .chain(_ => _.getExtraData())
                         .map<ApplicationData>(data => data[APPLICATION_DATA_KEY])
                         .map(data => data.appeal)
                         .unsafeCoerce();
 
+                    try{
+                        await parseFormData(request, response)
+                    } catch (error){
+                        if (error.message === 'File not supported'){
+                            return await that
+                                .renderUploadError(appeal, 'The selected file must be a TXT, DOC, PDF, JPEG or PNG');
+                        }
+                        else if(error.message === 'File too large'){
+                            return await that
+                                .renderUploadError(appeal, 'File size must be smaller than 4MB');
+                        }
+                    }
+
+                    if (!request.file){
+                        response.redirect(MOVED_TEMPORARILY, request.route.path);
+                        return;
+                    }else if(appeal.reasons.other.attachments &&
+                        appeal.reasons.other.attachments!.length >= maxNumberOfFiles){
+                        return await that
+                            .renderUploadError(appeal, 'You can only select up to 10 files at the same time');
+                    }
+
+                    let id: string;
+
+                    try {
+                        id = await that.fileTransferService.upload(request.file.buffer, request.file.originalname);
+                    } catch (err) {
+                        if (err.message === 'Request failed with status code 415') {
```

Comment:
I can see a pattern in most catch statements.
```
if (err.message === something) {
    doSomething()
    // OR
    returnSomething()
}
```

Could be worth extracting that pattern to avoid repetition